llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -10673,7 +10673,8 @@ LoopVectorizeResult LoopVectorizePass::runImpl(
   //Cheerp: currently JS does not support vector instructions,
   //they might be supported in the future though.
   const DataLayout &DL = F.getParent()->getDataLayout();
-  if (!DL.isByteAddressable()) {
+
+  if (!DL.isByteAddressable() && F.getSection() != "asmjs") {
```

Comment:
is this still needed? isn't it sufficient to return 0 vector registers in CheerpTargetTransform?

Reply:
Without the addition of the section, the code does not get auto-vectorized.
The check, when true, will stop loop vectorization, so without the check all functions skip vectorization, because ByteAddressable is always false.
So the addition only skips functions that are not tagged with 'asmjs'

Reply:
yeah but what about removing the whole if?

Reply:
Hard to tell. getNumberOfRegisters never actually returns 0, so every function is allowed to be vectorized.

Reply:
getNumberOfRegisters should return 0 for the vector class if simd is not enabled. And I think that it should be enough to stop loopvectorizer on genericjs code

Reply:
There's nothing in getNumberOfRegisters that differentiates between types of functions, so it doesn't filter between genericjs or not?

Reply:
true, in the end we didn't find a way to do it per function, now I remember